opentelemetry-api/src/opentelemetry/distributedcontext/__init__.py

```diff
@@ -11,3 +11,57 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+
+from contextlib import contextmanager
+import string
+import typing
+
+
+class EntryMetadata:
+    NO_PROPAGATION = 0
+    UNLIMITED_PROPAGATION = -1
+
+    def __init__(self, entry_ttl: int) -> None:
+        self.entry_ttl = entry_ttl
+
+
+class EntryKey(str):
+    def __new__(cls, value):
+        if any(c not in string.printable for c in value) or len(value) > 255:
+            raise ValueError("Invalid EntryKey", value)
+        return str.__new__(cls, value)
+
+
+class EntryValue(str):
+    def __new__(cls, value):
+        if any(c not in string.printable for c in value):
```

Comment:
I'm worried about this - `string.printable` contains many characters that we don't want. For example '\n'.